### DISCUSSION TITLE
Add an option to force the use of gpg keys

### DISCUSSION THREAD

**Discussion by ozoromo** [community/community#19678] [Code Security]
I would love for something like vigilant mode+ being added which auto rejects any cli pushes that are not signed with a verified key.
Protected branches are not that good of an alternative since new repos will be unlocked by default and re-protecting them is tedious.

**Comment by andrea-acampora** [comment]
Github allows you to sign commit with GPG.
First, you have to generate the keys and then you have to add the -S flag in the commit.
` git commit -S -m "your commit message"`
You can find more info at this [link](https://docs.github.com/en/authentication/managing-commit-signature-verification)
